src/prompt/variable.rs

```diff
@@ -131,106 +133,70 @@ fn quoted_str(input: &str) -> IResult<&str, String, ParsingError> {
-                                value('\u{0C}', char('f')),
-                                value('\\', char('\\')),
-                                value('/', char('/')),
-                                value('"', char('"')),
-                                value('\'', char('\'')),
-                            ))
-                        ),
-                        StringFragment::EscapedChar
-                    ),
-                    value(
-                        StringFragment::EscapedWS,
-                        preceded(char('\\'), nom::character::streaming::multispace1)
-                    ),
+        map_res(
+            recognize(
+                tuple((
+                    char(quote),
+                    move |str: &'a str| {
+                        let mut pos = 0;
+
+                        if str.is_empty() {
+                            return Err(Error(ParsingError::Incomplete));
+                        }
+
+                        if str.len() == 1 && str.chars().nth(0).unwrap() != quote {
+                            return Err(Error(ParsingError::Mistake {
+                                kind: None,
+                                description: "Missing end quote on string".to_string()
+                            }))
+                        }
+
+                        let chars = str.char_indices();
+
+                        let mut prev = None;
+
+                        for (i, c) in chars {
+                            if prev == None {
+                                prev = Some(c);
+                                continue;
+                            }
+
+                            if c == quote {
+                                // is it escaped?
+                                if prev.unwrap() == '\\' {
+                                    prev = Some(c);
+                                    continue;
+                                }
+
+                                // not escaped, its the end of the quoted string
+                                pos = i;
+                                break;
+                            }
+
+                            prev = Some(c)
+                        }
+
+                        // slice out the actual string part, and the remainder based on the 'pos'
+                        Ok((&str[pos..], &str[..pos]))
+                    },
+                    char(quote),
                 )),
-                String::new,
-                |mut string, fragment| {
-                    match fragment {
-                        StringFragment::Literal(s) => string.push_str(s),
-                        StringFragment::EscapedChar(c) => string.push(c),
-                        StringFragment::EscapedWS => {}
-                    }
-                    string
-                }
             ),
-            char(quote),
+            |s| unquote_string(s).map(|v| v.into()).context("Failed to unquote string")
```

Comment:
'Failed to unquote string' is pretty obscure for a user-facing message, I think

Reply:
It will include the message from the `Result` of `unquote_string` in the actual error message, I'll still change it to something more direct